tests/security/SingleParticipantWithSecurity/SingleParticipantWithSecurity.cpp

```diff
@@ -0,0 +1,326 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include <tests/DCPS/ConsolidatedMessengerIdl/MessengerTypeSupportC.h>
+#include <tests/DCPS/ConsolidatedMessengerIdl/MessengerTypeSupportImpl.h>
+
+#include <dds/DCPS/Marked_Default_Qos.h>
+#include <dds/DCPS/PublisherImpl.h>
+#include <dds/DCPS/SubscriberImpl.h>
+#include <dds/DCPS/Service_Participant.h>
+#include <dds/DCPS/WaitSet.h>
+#include <dds/DCPS/security/framework/Properties.h>
+#ifdef ACE_AS_STATIC_LIBS
+#  include <dds/DCPS/transport/rtps_udp/RtpsUdp.h>
+#  include <dds/DCPS/RTPS/RtpsDiscovery.h>
+#endif
+
+#include <ace/Get_Opt.h>
+#include <ace/Log_Msg.h>
+#include <ace/OS_NS_stdlib.h>
+#include <ace/OS_NS_unistd.h>
+
+#include <cstdlib>
+
+const char auth_ca_file_from_tests[] = "certs/identity/identity_ca_cert.pem";
+const char perm_ca_file_from_tests[] = "certs/permissions/permissions_ca_cert.pem";
+const char id_cert_file_from_tests[] = "certs/identity/test_participant_01_cert.pem";
+const char id_key_file_from_tests[] = "certs/identity/test_participant_01_private_key.pem";
+const char governance_file[] = "file:./governance_signed.p7s";
+const char permissions_file[] = "file:./permissions_1_signed.p7s";
+
+void append(DDS::PropertySeq& props, const char* name, const char* value, bool propagate = false)
+{
+  const DDS::Property_t prop = {name, value, propagate};
+  const unsigned int len = props.length();
+  props.length(len + 1);
+  props[len] = prop;
+}
+
+void read(DDS::DataReader_ptr reader)
+{
+  try {
+    Messenger::MessageDataReader_var message_dr =
+      Messenger::MessageDataReader::_narrow(reader);
+
+    if (!message_dr) {
+      ACE_ERROR((LM_ERROR,
+        ACE_TEXT("%N:%l: read()")
+        ACE_TEXT(" ERROR: _narrow failed!\n")));
+      return;
+    }
+
+    DDS::ReadCondition_var dr_rc = reader->create_readcondition(DDS::NOT_READ_SAMPLE_STATE,
+      DDS::ANY_VIEW_STATE,
+      DDS::ALIVE_INSTANCE_STATE);
+    DDS::WaitSet_var ws = new DDS::WaitSet;
+    ws->attach_condition(dr_rc);
+    std::set<int> instances;
+
+    DDS::ConditionSeq active;
+    const DDS::Duration_t max_wait = { 10, 0 };
+    DDS::ReturnCode_t ret = ws->wait(active, max_wait);
+
+    if (ret == DDS::RETCODE_TIMEOUT) {
+      ACE_ERROR((LM_ERROR,
+        ACE_TEXT("%N:%l: read()")
+        ACE_TEXT(" ERROR: reader timedout\n")));
+      return;
+    } else if (ret != DDS::RETCODE_OK) {
+      ACE_ERROR((LM_ERROR,
+        ACE_TEXT("%N:%l: read()")
+        ACE_TEXT(" ERROR: Reader: wait returned %d\n"), ret));
```

Comment:
```suggestion
        ACE_TEXT(" ERROR: Reader: wait returned %C\n"), OpenDDS::DCPS::retcode_to_string(ret)));
```
Need to include `DCPS_Utils.h` if this is changed.